File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.js
function solution(strArr) {
    const lengthMap = new Map();

    for (const str of strArr) {
        const len = str.length;
        lengthMap.set(len, (lengthMap.get(len) || 0) + 1);
    }

    return Math.max(...lengthMap.values());
}

File: 프로그래머스/0/181855. 문자열 묶기/문자열 묶기.java
class Solution {
    public int solution(String[] strArr) {
        // Map과 Array의 공간복잡도는 O(1)로 동일.
        // 그러나, Map의 내부 계산이 훨씬 무겁기 때문에 Array로 구현한다.
        
        int arr[] = new int[31];
        
        for (String s : strArr) {
            int idx = s.length();
            arr[idx]++;
        }
        
        int max = 0;
        for (int i : arr) {
            if (i > max) max = i;
        }
        
        return max;
    }
}